### PR TITLE
indi-full: Indi 3rdparty refactor

### PR DESCRIPTION
**Pull request by returntoreality**
## Description of changes

This is refactoring the indi 3rdparty drivers into their own derivations instead of having the firmware and 3rdparty derivations. This also sets the architecture support for each driver and assign the correct licences (before the unfree binary blobs from the 3rdparty repo were not marked as unfree). indi-full is now containing only free drivers and there is a new indi-full-nonfree which also contains the nonfree drivers.
I also added myself to the maintainers.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by returntoreality** [comment]
Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>56 packages built:</summary>
  <ul>
    <li>indi-full</li>
    <li>indi-full-nonfree</li>
    <li>indi3rdParty.indi-aagcloudwatcher-ng</li>
    <li>indi3rdParty.indi-aok</li>
    <li>indi3rdParty.indi-apogee</li>
    <li>indi3rdParty.indi-armadillo-platypus</li>
    <li>indi3rdParty.indi-asi</li>
    <li>indi3rdParty.indi-astroasis</li>
    <li>indi3rdParty.indi-astrolink4</li>
    <li>indi3rdParty.indi-astromechfoc</li>
    <li>indi3rdParty.indi-atik</li>
    <li>indi3rdParty.indi-avalon</li>
    <li>indi3rdParty.indi-avalonud</li>
    <li>indi3rdParty.indi-beefocus</li>
    <li>indi3rdParty.indi-bresserexos2</li>
    <li>indi3rdParty.indi-celestronaux</li>
    <li>indi3rdParty.indi-dreamfocuser</li>
    <li>indi3rdParty.indi-dsi</li>
    <li>indi3rdParty.indi-duino</li>
    <li>indi3rdParty.indi-eqmod</li>
    <li>indi3rdParty.indi-ffmv</li>
    <li>indi3rdParty.indi-fishcamp</li>
    <li>indi3rdParty.indi-fli</li>
    <li>indi3rdParty.indi-gige</li>
    <li>indi3rdParty.indi-gphoto</li>
    <li>indi3rdParty.indi-gpsd</li>
    <li>indi3rdParty.indi-gpsnmea</li>
    <li>indi3rdParty.indi-inovaplx</li>
    <li>indi3rdParty.indi-limesdr</li>
    <li>indi3rdParty.indi-maxdomeii</li>
    <li>indi3rdParty.indi-mgen</li>
    <li>indi3rdParty.indi-mi</li>
    <li>indi3rdParty.indi-nexdome</li>
    <li>indi3rdParty.indi-nightscape</li>
    <li>indi3rdParty.indi-nut</li>
    <li>indi3rdParty.indi-ocs</li>
    <li>indi3rdParty.indi-orion-ssg3</li>
    <li>indi3rdParty.indi-pentax</li>
    <li>indi3rdParty.indi-playerone</li>
    <li>indi3rdParty.indi-qhy</li>
    <li>indi3rdParty.indi-qsi</li>
    <li>indi3rdParty.indi-rolloffino</li>
    <li>indi3rdParty.indi-rtklib</li>
    <li>indi3rdParty.indi-sbig</li>
    <li>indi3rdParty.indi-shelyak</li>
    <li>indi3rdParty.indi-spectracyber</li>
    <li>indi3rdParty.indi-starbook</li>
    <li>indi3rdParty.indi-starbook-ten</li>
    <li>indi3rdParty.indi-svbony</li>
    <li>indi3rdParty.indi-sx</li>
    <li>indi3rdParty.indi-talon6</li>
    <li>indi3rdParty.indi-toupbase</li>
    <li>indi3rdParty.indi-webcam</li>
    <li>indi3rdParty.indi-weewx-json</li>
    <li>kstars</li>
    <li>phd2</li>
  </ul>
</details>

**Review by sheepforce — APPROVED** [review]
Uh wow, this looks so much better and cleaner! :heart_eyes: :sheep: Thank you!  Do you have any idea if r-ryantm updates work on them correctly now?

**Comment by returntoreality** [comment]
Thanks! Regarding the r-ryantm bot, I don't think it fixes it. I think indilib needs a `passthru.updateScript` that also updates the indi-3rdparty repo. I wanted to look into this after these changes are merged. Since the 23.05 release is soon no breaking changes are supposed to be merged (I think this would probably count as one).

**Comment by returntoreality** [comment]
I think it can be merged now, I rebased it to the current master.

**Review by sheepforce — COMMENTED** [review]
Unfortunately there seems to be a build failure in  ofBorg. Otherwise looks very nice!

**Review by JohnRTitor — COMMENTED** [review]
Hi, could you rebase this? and this seems to fail on aarch64-linux?

**Review by sheepforce — CHANGES_REQUESTED** [review]
This looks like a very good idea overall, thank you! 

Two (hopefully) minor things left:

  * There is a merge conflict to be resolved
  * It fails on aarch64-linux as it tries to compile with Raspberry Pi GPIO support. Ofborg has the error log here https://logs.ofborg.org/?key=nixos/nixpkgs.306650&attempt_id=c546a1ea-0537-4f2e-b08c-f8d3dd622a59 and it looks like `pigpiod_if2.h` is missing. I'm not sure whether this is because the builder is not a RPi or if there is actually a dependency not declared?

I think we should merge this, after these two are resolved :)

File: pkgs/by-name/in/indi-with-drivers/package.nix
Comment (JohnRTitor) [inline pkgs/by-name/in/indi-with-drivers/package.nix:18]: It makes sense to invert the condition and use `lib.optionalString` here.

Comment (returntoreality) [reply pkgs/by-name/in/indi-with-drivers/package.nix]: Indeed, that is nicer, thanks for the suggestion!

**Comment by returntoreality** [comment]
Thanks for the reviews! 
> Two (hopefully) minor things left:
>     * There is a merge conflict to be resolved

This is solved now, moved the changelog entry to a random position so hopefully it won't have a conflict anytime soon.

> 
>     * It fails on aarch64-linux as it tries to compile with Raspberry Pi GPIO support. Ofborg has the error log here https://logs.ofborg.org/?key=nixos/nixpkgs.306650&attempt_id=c546a1ea-0537-4f2e-b08c-f8d3dd622a59 and it looks like `pigpiod_if2.h` is missing. I'm not sure whether this is because the builder is not a RPi or if there is actually a dependency not declared?

I disabled the driver, this apparently relies on the pigpio library/service which is not avalable in nixpkgs currently.
 
> I think we should merge this, after these two are resolved :)

That would be great!

**Pull request merged**
